app/models/manageiq/providers/autosde/storage_manager/ems_refresh_workflow.rb

```diff
@@ -22,6 +28,29 @@ def poll_native_task
     signal(:abort, err.message, "error")
   end
 
+  def poll_refresh_task
+    opts = {
+      :query_params => {
+        :object_id => options[:native_object_id]
+      }
+    }
+    tasks = autosde_client.JobApi.jobs_get(opts)
+    raise _('No refresh task found') if tasks.count < 1
+    task = tasks.detect { |t| t.task_name.match("update_system") }
+    raise _('No refresh task found') if task.nil?
+    case task.status
```

Comment:
Two minor things, 1. you want `match?` not `match` because all you care about is the boolean return and 2. rubocop will warn about not having a newline after the raise
```suggestion
    tasks = autosde_client.JobApi.jobs_get(opts)
    raise _('No refresh task found') if tasks.count < 1

    task = tasks.detect { |t| t.task_name.match?("update_system") }
    raise _('No refresh task found') if task.nil?

    case task.status
```

Also just checking that the `task_name` isn't literally `"update_system"`?  If so you could do a `t.task_name == "update_system"`

Reply:
Okay I see from above the name is `autosde_site_manager.engine.tasks.update_system` so good with `.match?`